### DISCUSSION TITLE
GraphQL endpoint for test store?

### DISCUSSION THREAD

**Discussion by rstml** [swellstores/community#385] [Help] [answered]
Each store on Swell has test and live mode. Given that they can have different models, there should be a way to query each one through GraphQL.

However, I could only find GraphQL endpoint for the live store, which looks like ` https://mystore.swell.store/graphql`.

Is there an endpoint for the test store?

**Comment by sags95** [comment, marked as answer]
@rstml The endpoint would remain the same, but you would pass in the test mode public key denoted with `pk_test_***`. This can be found in Developers > API Keys

**Reply by rstml** [reply]
In that case, I think there must be some problem/bug.

I created a custom content field under `products` in test and GraphQL schema didn't reflect this until I pushed the same field to live version.

**Reply by rstml** [reply]
Moreover, test model explorer admin UI (`/admin/test/model-explorer`) also didn't reflect this until I added it for live mode.

One more bug, in the test model explorer UI link to console `</>` takes me to live version. Confusing.

<img width="362" alt="Screenshot 2023-09-21 at 16 44 34" src="https://github.com/swellstores/community/assets/589231/57f00226-27be-4112-9681-652954d63ee0">

**Reply by logeshswell** [reply]
You're right, model explorer and GraphQL schema replicates the contents from the live env in test env.

I will file this issue and keep you posted if I get some updates.
